_episodes_rmd/XX-bioconductor-vcfr.Rmd

```diff
@@ -0,0 +1,90 @@
+- "Installing packages from Bioconductor requires a new method, since it is not compatible with the `install.packages()` function used for CRAN."
+- "Check Bioconductor to see if there is a package relevent to your analysis before writing code yourself."
+source: Rmd
+---
+
+```{r chunk_options, include=FALSE}
+source("../bin/chunk-options.R")
+
+# TODO: This needs to be updated when the episode number is finalized
+knitr_fig_path("XX-")
+```
+
+<!-- example of rendered lesson https://datacarpentry.org/genomics-r-intro/02-r-basics/index.html -->
+
+## Installing packages from somewhere else besides CRAN?
+
+In some cases, you may want to use a specialized package that is not hosted on [CRAN](https://cran.r-project.org/) (the Comprehensive R Archive Network). This may be because the package is so new that it hasn't yet been submitted to CRAN, or it could be that it is on a focal topic that has an alternative repository. One major example of an alternative repository source is [Bioconductor](https://bioconductor.org/), which has a mission of "promot[ing] the statistical analysis and comprehension of current and emerging high-throughput biological assays." This means that many if not all of the packages available on Bioconductor are focused on the analysis of biological data, and that it can be a great place to look for tools to help you analyze your -omics datasets!
+
+## So how do I use it?
+
+Since access to the [Bioconductor](https://bioconductor.org/) repository is not built in to base R 'out of the box', there are a couple steps needed to install packages from this alternative source. We will work through the steps (only 2!) to install a package to help with the VCF analysis we are working on, but you can use the same approach to install any of the many thousands of available packages.
+
+![screenshot of bioconductor homepage](fig/bioconductor_website_screenshot.jpg)
+
+## First, install the `BiocManager` package
+
+The first step is to install a package that *is* on CRAN, `BiocManager`. This package will allow us to use it to install packages from Bioconductor. You can think of Bioconductor kind of like an alternative app store for your phone, except instead of apps you are installing packages, and instead of your phone it's your local R package library.
+
+```{r install-biocmanager, eval = FALSE}
+# install the BiocManager from CRAN using the base R install.packages() function
+install.packages("BiocManager")
+```
+
+To check if this worked (and also so you can make a note of the version for reproducibility purposes), you can run `BiocManager::version()` and it should give you the version number.
+
+```{r check-bioconductor-version, eval = FALSE}
+# to make sure it worked, check the version
+BiocManager::version()
+```
+
+# Second, install the vcfR package from Bioconductor using `BiocManager`
+
+```{r install-vcfR, eval = FALSE}
+# install the vcfR package from bioconductor using BiocManager::install()
+BiocManager::install("vcfR")
```

Comment:
I saw this and I want to point out that you might want to use a different package to demonstrate this since vcfR is on CRAN, and not Bioconductor (source: I worked with the developer).

Reply:
@zkamvar -- that's a good point. We need to show installing `vcfR` since it gets used later in the sequence, but it would be better to also show a bioconductor-only package. Do you have any suggestions of vcf-related packages on bioconductor that you'd recommend as an example?

Reply:
Luckily, the author has been curating a list of other software that do similar things: https://github.com/knausb/vcfR#software-that-produce-vcf-files

VariantAnnotation seems to be another popular VCF package: https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html